torch/optim/adam.py

```diff
@@ -152,7 +154,8 @@ def step(self, closure=None):
 
                     if group['amsgrad']:
                         max_exp_avg_sqs.append(state['max_exp_avg_sq'])
-
+                    if group['differentiable']:
+                        assert not state['step'].requires_grad
```

Comment:
nit make this a RuntimeError and add a nice error message.

Reply:
done!